src/main/java/org/blockchain_innovation/factom/identiy/did/DIDVersion.java

```diff
@@ -23,20 +24,26 @@
     }
 
     public void assertFactomMethod(String didUrl) {
-        if (!method.equals(getDid(didUrl).getMethod())) {
+        String didUrlMethod;
+        try {
+            didUrlMethod = getDid(didUrl).getMethod();
+        } catch (ParserException e){
+            throw new DIDRuntimeException("Could not parse DID URL: " + didUrl);
+        }
+        if (!method.equals(didUrlMethod)) {
             throw new DIDRuntimeException("Method of DID URL is not supported by this version of Factom DIDs: " + didUrl);
         }
     }
 
-    public String getMethodSpecificId(String didUrl) {
+    public String getMethodSpecificId(String didUrl) throws ParserException {
```

Comment:
Hmzz, I rather see a runtime exception as opposed to a checked exception for these types of problems

Reply:
Yes, good point. Updated with existing parse exception in `DIDRuntimeException`.